src/MiniCover/Reports/Clover/CloverReport.cs

```diff
@@ -0,0 +1,198 @@
+        {
+            return result.Assemblies.Select(assembly => new XElement(
+                XName.Get("package"),
+                new XAttribute(XName.Get("name"), assembly.Name),
+                CreateMetricsElement(CountPackageMetrics(assembly, hits)),
+                CreateFilesElement(assembly, hits)
+            ));
+        }
+
+        private static IEnumerable<XElement> CreateFilesElement(InstrumentedAssembly assembly, IDictionary<int, int> hits)
+        {
+            return assembly.SourceFiles.Select(file => new XElement(
+                XName.Get("file"),
+                new XAttribute(XName.Get("name"), Path.GetFileName(file.Key)),
+                new XAttribute(XName.Get("path"), file.Key),
+                CreateMetricsElement(CountFileMetrics(file.Value, hits)),
+                CreateClassesElement(file.Value.Instructions, hits),
+                CreateLinesElement(file.Value.Instructions, hits)
+            ));
+        }
+
+        private static IEnumerable<XElement> CreateClassesElement(ICollection<InstrumentedInstruction> instructions, IDictionary<int, int> hits)
+        {
+            return instructions
+                .GroupBy(instruction => instruction.Class)
+                .Select(classes => new XElement(
+                    XName.Get("class"),
+                    new XAttribute(XName.Get("name"), classes.Key),
+                    CreateMetricsElement(CountMetrics(classes.AsEnumerable(), hits))
+                ));
+        }
+
+        private static IEnumerable<XElement> CreateLinesElement(ICollection<InstrumentedInstruction> instructions, IDictionary<int, int> hits)
+        {
+            return instructions
+                .Select(instruction => new XElement(
+                    XName.Get("line"),
+                    new XAttribute(XName.Get("num"), instruction.StartLine),
+                    new XAttribute(XName.Get("count"), hits.TryGetValue(instruction.Id, out int count) ? count : 0),
+                    new XAttribute(XName.Get("type"), "stmt")
+                ));
+        }
+
+        private static XElement CreateMetricsElement(CloverCounter counter)
+        {
+            var element = new XElement(
+                XName.Get("metrics"),
+                new XAttribute(XName.Get("statements"), counter.Statements),
+                new XAttribute(XName.Get("coveredstatements"), counter.CoveredStatements),
+                new XAttribute(XName.Get("conditionals"), counter.Conditionals),
+                new XAttribute(XName.Get("coveredconditionals"), counter.CoveredConditionals),
+                new XAttribute(XName.Get("methods"), counter.Methods),
+                new XAttribute(XName.Get("coveredmethods"), counter.CoveredMethods),
+                new XAttribute(XName.Get("elements"), counter.Elements),
+                new XAttribute(XName.Get("coveredelements"), counter.CoveredElements)
+            );
+
+            if(counter.Lines > 0)
```

Comment:
any will be clearer

Reply:
it's faster than 'any'

Reply:
sorry, 'lines' is of type Int. We can not use 'any'